test/test_device.py

```diff
@@ -609,3 +623,44 @@ def testStringAsBool(self):
         self.device.declare({'Status': {'type': 'string'}})
         self.device.publish_data({'Status': True})
         self.api.publish_data.assert_called_with({'Status': 'true'})
+
+    def testLocation(self):
+        location = {'lat': 37.89, 'lng': 75.43}
+        self.device.declare({'Pos': {'type': 'location'}})
+        self.device.publish_data({'Pos': location})
+        self.api.publish_data.assert_called_with({'Pos': location})
+
+    def testLocation_Filtering(self):
+        obj = {'some': 'foo', 'lng': 75.43, 'lat': 37.89, 'other': 42}
```

Comment:
I don't think it's required to filter the dict

Reply:
See above.